lib/client/api.go

```diff
@@ -244,6 +248,19 @@ type Config struct {
 
 	// BindAddr is an optional host:port to bind to for SSO redirect flows
 	BindAddr string
+
+	// ServerFeatures lists additional features supported by the server.
+	//
+	// This field is populated based on the settings received from the server
+	// as a part of the ping response.
+	ServerFeatures []string
+
+	// Docker is responsible for configuring access to Docker registy that
+	// may be optionally provided by the server.
+	Docker configurator.Configurator
```

Comment:
If you don't really care about configurators types (as they are the same interface) why not have a generic list of them